test/integration/provider/gcp.go

```diff
@@ -0,0 +1,200 @@
+package provider
+
+/**
+	Orphaned Resources
+	- VMs:
+		Describe instances with specified tag name:<cluster-name>
+		Report/Print out instances found
+		Describe volumes attached to the instance (using instance id)
+		Report/Print out volumes found
+		Delete attached volumes found
+		Terminate instances found
+	- Disks:
+		Describe volumes with tag status:available
+		Report/Print out volumes found
+		Delete identified volumes
+**/
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+
+	"net/http"
+
+	api "github.com/gardener/machine-controller-manager-provider-gcp/pkg/api/v1alpha1"
+	providerDriver "github.com/gardener/machine-controller-manager-provider-gcp/pkg/gcp"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/driver"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func GetMachines(machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
+	var machines []string
+	var sPI providerDriver.PluginSPIImpl
+	driverprovider := providerDriver.NewGCPPlugin(&sPI)
+	machineList, err := driverprovider.ListMachines(context.TODO(), &driver.ListMachinesRequest{
+		MachineClass: machineClass,
+		Secret:       &v1.Secret{Data: secretData},
+	})
+	if err != nil {
+		return nil, err
+	} else if len(machineList.MachineList) != 0 {
+		fmt.Printf("\nAvailable Machines: ")
+		for _, machine := range machineList.MachineList {
+			fmt.Printf("%s\n", machine)
+			machines = append(machines, machine)
+		}
+	}
+	return machines, nil
+}
+
+// GetInstancesWithTag describes the instance with the specified tag
+func GetInstancesWithTag(ctx context.Context, svc *compute.Service, searchTagName string, machineClass *v1alpha1.MachineClass, secretData map[string][]byte) ([]string, error) {
```

Comment:
Just like above, unexported function would be good. Please do the same for rest of the functions in this file.